lambda-runtime/src/AWS/Lambda/Runtime.hs

```diff
@@ -0,0 +1,40 @@
+module AWS.Lambda.Runtime
+  ( run
+  )
+where
+
+import AWS.Prelude
+import Control.Monad.Except (runExceptT)
+
+import qualified AWS.Lambda.Client as Client
+import qualified Data.Aeson        as JSON
+
+run
+  :: forall env m . (MonadCatch m, MonadIO m)
+  => env
+  -> (JSON.Value -> RIO env JSON.Value)
+  -> m ()
+run env lambdaFn = forever . runtimeLoop $ (runRIO env . lambdaFn)
+
+runtimeLoop
+  :: forall m . (MonadIO m, MonadCatch m)
+  => (JSON.Value -> m JSON.Value)
+  -> m ()
+runtimeLoop fn =
+  either throwM runFunction =<< liftIO (runExceptT Client.getHttpConfig)
```

Comment:
@epicallan Just to confirm. This fixes our main design issue with `hal` as it would crash the container (assuming outer call side of `run` does not re-start the loop).

Reply:
Yes, this would crash the container assuming the outer call side of `run` doesn't attempt to catch any error and restart the loop.